File: Chapter 4/Newton.cpp
#include <iostream>
using namespace std;

//This program will take the user input of a number and find its square
//It will make 20 guesses based on Newtons Method
//It will then output the square root of a number 

double squareroot(double n) {

    double root = n / 2;

    for (int i = 0; i < 20; i++) {
        root = (0.5) * (root + (n / root));
    }

    return root;
}

int main() {
    double userInput;
    cout << "Enter the number to calculate its square root: ";
    cin >> userInput;

    cout << "The square root of " << userInput << " is: " << squareroot(userInput) << endl;

    return 0;

}

File: Chapter 4/Newton.py
#This program is the Python version of Newtons Method

def squareroot(n):
    root = n / 2.0
    for i in range (20):
        root = 0.5 * (root + n / root)
    return root

if __name__ == "__main__":
    while True:
            user_input = input("Enter a number to find the square root or type 'exit' to quit: ")
            if user_input.lower() == "exit":
                print("Goodbye!")
                break

            try:
                 number = float(user_input) #turn the users input into a float
                 print(f"The square root of {number} is approximately {squareroot(number)}")

            except ValueError:
                 print("Invalid input, please enter a valid number")

            #break


#The while True statement will keep the program working until the break statement is executed
#If break is put in line 23 the program will not continue after the user enters an invalid input
